crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -243,38 +245,53 @@ impl<'this, TX: DbTxMut<'this> + DbTx<'this>> DatabaseProvider<'this, TX> {
 
         let storage_changeset =
             self.get_or_take::<tables::StorageChangeSet, UNWIND>(storage_range)?;
-        let account_changeset = self.get_or_take::<tables::AccountChangeSet, UNWIND>(range)?;
+        let account_changeset =
+            self.get_or_take::<tables::AccountChangeSet, UNWIND>(range.clone())?;
 
         // iterate previous value and get plain state value to create changeset
         // Double option around Account represent if Account state is know (first option) and
         // account is removed (Second Option)
 
-        let mut state: BundleStateInit = HashMap::new();
-
+        let mut state = HashMap::new();
+        let mut bundle_builder = BundleBuilder::new(range);
         // This is not working for blocks that are not at tip. as plain state is not the last
         // state of end range. We should rename the functions or add support to access
         // History state. Accessing history state can be tricky but we are not gaining
         // anything.
         let mut plain_accounts_cursor = self.tx.cursor_write::<tables::PlainAccountState>()?;
         let mut plain_storage_cursor = self.tx.cursor_dup_write::<tables::PlainStorageState>()?;
 
-        let mut reverts: RevertsInit = HashMap::new();
-
         // add account changeset changes
         for (block_number, account_before) in account_changeset.into_iter().rev() {
             let AccountBeforeTx { info: old_info, address } = account_before;
             match state.entry(address) {
                 hash_map::Entry::Vacant(entry) => {
                     let new_info = plain_accounts_cursor.seek_exact(address)?.map(|kv| kv.1);
                     entry.insert((old_info, new_info, HashMap::new()));
+                    bundle_builder = bundle_builder.state_original_account_info(
+                        address,
+                        into_revm_acc(old_info.unwrap_or_default()),
+                    );
+                    bundle_builder = bundle_builder.state_present_account_info(
+                        address,
+                        into_revm_acc(new_info.unwrap_or_default()),
```

Comment:
Given the introduction of these, shouldn't `let mut state` be unnecessary, and as a result all the `match state.entry` calls? I would think that the bundle builder would be able to recognize where to put these internally, otherwise we're now double counting